debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -606,44 +675,49 @@ public void raiseError(Exception e) {
                     }
                 } catch (Throwable t) {
                     fail("Error while trying to run connector class '" + connectorClassName + "'", t);
-                    return;
                 } finally {
                     // Close the offset storage and finally the connector ...
                     try {
                         offsetStore.stop();
                     } finally {
                         connector.stop();
+                        connectorCallback.ifPresent(ConnectorCallback::connectorStopped);
```

Comment:
Should we handle the case when `connector.stop()` throws an error? Should line #684 be in a finally block?